From 5d14551e4a0a1265fd2ffc0c28bf88c7cce0ad96 Mon Sep 17 00:00:00 2001
From: ChonDoit <thphantomblog@gmail.com>
Date: Sun, 2 Jul 2023 23:56:08 +0000
Subject: [PATCH] Add sysfs_battery_supply to duplicated list of Samsung
 SEPolicy typealias

Saw on A73 5G

[    1.902951] [7:         secilc:  395]  secilc (395) used greatest stack depth: 10464 bytes left
[    1.903132] [3:           init:    1]  init: /system/bin/secilc: linker: Warning: failed to find generated linker configuration from "/linkerconfig/ld.config.txt"
[    1.903160] [3:           init:    1]  init: /system/bin/secilc: WARNING: linker: Warning: failed to find generated linker configuration from "/linkerconfig/ld.config.txt"
[    1.903180] [3:           init:    1]  init: /system/bin/secilc: Re-declaration of typealias sysfs_battery_supply
[    1.903200] [3:           init:    1]  init: /system/bin/secilc: Previous declaration of type at /vendor/etc/selinux/vendor_sepolicy.cil:909
[    1.903219] [3:           init:    1]  init: /system/bin/secilc: Bad typealias declaration at /vendor/etc/selinux/vendor_sepolicy.cil:909
[    1.903235] [3:           init:    1]  init: /system/bin/secilc: Failed to build AST
[    1.903251] [3:           init:    1]  init: /system/bin/secilc: Failed to compile cildb: -1
[    1.904728] [3:           init:    1]  init: /system/bin/secilc exited with status 255
[    1.904835] [3:           init:    1]  init: Unable to open SELinux policy
[    1.905637] [3:           init:    1]  init: InitFatalReboot: signal 6
[    1.915980] [3:           init:    1]  init: #00 pc 000000000012cafc  /system/bin/init (android::init::InitFatalReboot(int)+104)
[    1.916003] [3:           init:    1]  init: #01 pc 00000000000c0cb4  /system/bin/init (android::init::InitAborter(char const*)+48)
[    1.916048] [3:           init:    1]  init: #02 pc 0000000000016ea8  /system/lib64/libbase.so (android::base::SetAborter(std::__1::function<void (char const*)>&&)::$_3::__invoke(char const*)+80)
[    1.916070] [3:           init:    1]  init: #03 pc 0000000000016450  /system/lib64/libbase.so (android::base::LogMessage::~LogMessage()+352)
[    1.916090] [3:           init:    1]  init: #04 pc 00000000001324b4  /system/bin/init (android::init::SetupSelinux(char**)+9648)
[    1.916109] [3:           init:    1]  init: #05 pc 00000000000ae17c  /system/bin/init (main+284)
[    1.916128] [3:           init:    1]  init: #06 pc 0000000000090ad0  /system/lib64/bootstrap/libc.so (__libc_init+96)
[    1.916144] [3:           init:    1]  init: Reboot ending, jumping to kernel
[    1.916211] [3:           init:    1]  (sec_debug_update_restart_reason) reboot cmd : recovery
---
 libsepol/cil/src/cil_build_ast.c   | 1 +
 libsepol/cil/src/cil_resolve_ast.c | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/libsepol/cil/src/cil_build_ast.c b/libsepol/cil/src/cil_build_ast.c
index 61c8864b..81f83e7b 100644
--- a/libsepol/cil/src/cil_build_ast.c
+++ b/libsepol/cil/src/cil_build_ast.c
@@ -132,6 +132,7 @@ int cil_add_decl_to_symtab(struct cil_db *db, symtab_t *symtab, hashtab_key_t ke
 			if(
 				strcmp(key, "sysfs_usb_supply") == 0 ||
 				strcmp(key, "hostapd") == 0 ||
+				strcmp(key, "sysfs_battery_supply") == 0 ||
 				strcmp(key, "rpmb_device") == 0) {
 					cil_log(CIL_ERR, "Ignoring...");
 			} else {
diff --git a/libsepol/cil/src/cil_resolve_ast.c b/libsepol/cil/src/cil_resolve_ast.c
index 87db4f81..d11bbbda 100644
--- a/libsepol/cil/src/cil_resolve_ast.c
+++ b/libsepol/cil/src/cil_resolve_ast.c
@@ -520,6 +520,7 @@ int cil_resolve_aliasactual(struct cil_tree_node *current, void *extra_args, enu
 		if(
 			strcmp(alias_datum->name, "hostapd") == 0 ||
 			strcmp(alias_datum->name, "sysfs_usb_supply") == 0 ||
+			strcmp(alias_datum->name, "sysfs_battery_supply") == 0 ||
 			strcmp(alias_datum->name, "rpmb_device") == 0)
 				rc = 0;
 			else
@@ -568,6 +569,7 @@ int cil_resolve_alias_to_actual(struct cil_tree_node *current, enum cil_flavor f
 		if(
 			strcmp(a1->datum.name, "hostapd") == 0 ||
 			strcmp(a1->datum.name, "sysfs_usb_supply") == 0 ||
+			strcmp(a1->datum.name, "sysfs_battery_supply") == 0 ||
 			strcmp(a1->datum.name, "rpmb_device") == 0)
 				return SEPOL_OK;
 		return SEPOL_ERR;
-- 
2.34.1

